com.unity.render-pipelines.universal/ShaderLibrary/MetaInput.hlsl

```diff
@@ -1,70 +1,85 @@
+    float4 positionCS   : SV_POSITION;
+    float2 uv           : TEXCOORD0;
 };
 
-float4 MetaVertexPosition(float4 positionOS, float2 uv1, float2 uv2, float4 uv1ST, float4 uv2ST)
+MetaVaryings UniversalMetaVertexPosition(float4 positionOS, float2 uv0, float2 uv1, float2 uv2)
 {
-    if (unity_MetaVertexControl.x)
-    {
-        positionOS.xy = uv1 * uv1ST.xy + uv1ST.zw;
-        // OpenGL right now needs to actually use incoming vertex position,
-        // so use it in a very dummy way
-        positionOS.z = positionOS.z > 0 ? REAL_MIN : 0.0f;
-    }
-    if (unity_MetaVertexControl.y)
+    MetaVaryings ret = (MetaVaryings)0;
+    ret.positionCS = UnityMetaVertexPosition(positionOS.xyz, uv1, uv2, unity_LightmapST, unity_DynamicLightmapST);
+
+#ifdef EDITOR_VISUALIZATION
+    if (unity_VisualizationMode == EDITORVIZ_TEXTURE)
+        ret.VizUV = UnityMetaVizUV(unity_EditorViz_UVIndex, uv0, uv1, uv2, unity_EditorViz_Texture_ST);
+    else if (unity_VisualizationMode == EDITORVIZ_SHOWLIGHTMASK)
     {
-        positionOS.xy = uv2 * uv2ST.xy + uv2ST.zw;
-        // OpenGL right now needs to actually use incoming vertex position,
-        // so use it in a very dummy way
-        positionOS.z = positionOS.z > 0 ? REAL_MIN : 0.0f;
+        ret.VizUV = uv1 * unity_LightmapST.xy + unity_LightmapST.zw;
+        ret.LightCoord = mul(unity_EditorViz_WorldToLight, float4(TransformObjectToWorld(positionOS), 1));
     }
-    return TransformWorldToHClip(positionOS.xyz);
+#endif
+    return ret;
 }
 
-half4 MetaFragment(MetaInput input)
+MetaVaryings UniversalVertexMeta(Attributes input)
 {
-    half4 res = 0;
-    if (unity_MetaFragmentControl.x)
-    {
-        res = half4(input.Albedo, 1.0);
+    MetaVaryings output;
+    output = UniversalMetaVertexPosition(input.positionOS, input.uv0, input.uv1, input.uv2);
+    return output;
+}
 
-        // Apply Albedo Boost from LightmapSettings.
-        res.rgb = clamp(PositivePow(res.rgb, saturate(unity_OneOverOutputBoost)), 0, unity_MaxOutputValue);
-    }
-    if (unity_MetaFragmentControl.y)
-    {
-        half3 emission;
-        if (unity_UseLinearSpace)
-            emission = input.Emission;
-        else
-            emission = LinearToSRGB(input.Emission);
+half4 UniversalFragmentMeta(MetaVaryings fragIn, UnityMetaInput metaInput)
+{
+#ifdef EDITOR_VISUALIZATION
+    metaInput.VizUV = fragIn.VizUV;
+    metaInput.LightCoord = fragIn.LightCoord;
+#endif
 
-        res = half4(emission, 1.0);
-    }
-    return res;
+    return UnityMetaFragment(metaInput);
 }
 
+//LWRP -> Universal Backwards Compatibility
+MetaVaryings VaryingsToMetaVaryings(Varyings input)
+{
+    MetaVaryings ret = (MetaVaryings)0;
+    ret.positionCS = input.positionCS;
+    ret.uv = input.uv;
+    return ret;
+}
+
+Varyings LightweightVertexMeta(Attributes input)
+{
+    Varyings ret;
+    MetaVaryings mRet = UniversalVertexMeta(input);
+    ret.positionCS = mRet.positionCS;
+    ret.uv = mRet.uv;
+    return ret;
+}
```

Comment:
This is probably merge conflict? It should be removed as we removed LWRP upgrade tooling here: https://github.com/Unity-Technologies/Graphics/pull/3051/files#diff-42c6b6ac3d960a42860bdc7f7b585ebe5c4f1f2580fa2fde5e805d33e05f92e3L49-L59